airflow/models/deadlines.py

```diff
@@ -0,0 +1,100 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING
+
+import sqlalchemy_jsonfield
+from sqlalchemy import Column, DateTime, Index, Integer, String
+
+from airflow.models.base import Base, StringID
+from airflow.settings import json
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.session import NEW_SESSION, provide_session
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+
+class Deadlines(LoggingMixin):
```

Comment:
I guess it is a question of style and personal preference why having 2 classes? Why not moving this method to `DeadlinesModel `?

Reply:
I was following the example of DAG/DagModel, but I guess I can consolidate them.  Let me think it through real quick and maybe I'll do that.

Reply:
It looks like the DAG example was the exception, not the rule.  I think I'll consolidate them both into Deadline.  That seems to be the more common convention.